src/uu/tr/src/operation.rs

```diff
@@ -577,15 +577,25 @@ impl SymbolTranslator for SqueezeOperation {
     }
 }
 
-pub fn translate_input<T, R, W>(input: &mut R, output: &mut W, mut translator: T)
+pub fn translate_input<T, R, W>(input: &mut R, output: &mut W, mut translator: T) -> UResult<()>
 where
     T: SymbolTranslator,
     R: BufRead,
     W: Write,
 {
     let mut buf = Vec::new();
     let mut output_buf = Vec::new();
-    while let Ok(length) = input.read_until(b'\n', &mut buf) {
+    loop {
+        let length = match input.read_until(b'\n', &mut buf) {
+            Ok(0) => break, // EOF reached
```

Comment:
This line makes the `break` on line 600 unreachable. So I would either remove this line (and probably you can get rid of the `match`, too) or remove the `if` and the `if`-case on lines 599 - 601.

Reply:
better this way, no ?